be/src/main/java/todolist/repository/CardMemoryRepository.java

```diff
@@ -22,17 +24,25 @@ public Card save(Card card) {
     }
 
     @Override
-    public Optional<Card> findById(Long id) {
-        return Optional.empty();
+    public Card findById(Long id) {
+        return store.stream().filter(card -> card.getId() == id).findAny().orElseThrow(()-> new IllegalArgumentException());
```

Comment:
라인 indentation을 읽기 쉬운 형태로 가져갑시다. 각각의 Stream 내부 함수 별로 쪼개면 됩니다.
```suggestion
        return store.stream().filter(card -> card.getId() == id)
                             .findAny()
                             .orElseThrow(()-> new IllegalArgumentException());
```

Reply:
**도전 과제**: Exception을 throw하는데, 현재 시점에서는 어떠한 오류인지 클라이언트 입장에서는 알 수 있는 길이 없습니다. 명시적으로 exception을 던져주기 위해서는 어떻게 해야 할까요? 가장 간단하게는 원인 메시지를 명시하는 방법이 있을 것이고, 더 나아가서는 Custom Exception을 만들어 메시지를 리턴할 수도 있겠고 이에 따라서 HTTP Code를 별도로 부여하여 클라이언트에 내려줄 수도 있을 것 같습니다.